frontend/src/components/explorer/gemBrowser/Compartment.vue

```diff
@@ -105,36 +105,17 @@ export default {
       return l.join('');
     },
   },
-  watch: {
-    /* eslint-disable quote-props */
-    '$route': async function watchSetup() {
-      if (this.$route.path.includes('/gem-browser/') && this.$route.path.includes('/compartment/')) {
-        if (this.cName !== this.$route.params.id) {
-          await this.setup();
-        }
-      }
-    },
-  },
   async beforeMount() {
-    await this.setup();
-  },
-  methods: {
-    async setup() {
-      this.cName = this.$route.params.id;
-      await this.load();
-    },
-    async load() {
-      this.showLoader = true;
-      try {
-        const payload = { model: this.model.database_name, id: this.cName };
-        await this.$store.dispatch('compartments/getCompartmentSummary', payload);
-        this.componentNotFound = false;
-        this.showLoader = false;
-      } catch {
-        this.componentNotFound = true;
-        document.getElementById('search').focus();
-      }
-    },
+    this.cName = this.$route.params.id;
+    try {
+      const payload = { model: this.model.database_name, id: this.cName };
+      await this.$store.dispatch('compartments/getCompartmentSummary', payload);
+      this.componentNotFound = false;
+      this.showLoaderMessage = '';
```

Comment:
Don't remember how `v-if` works now but does a `''` count as false?

Reply:
Indeed, just wanted to be type-consistent with the new usage as string instead of boolean.